src/wreadinput/device.py

```diff
@@ -33,12 +33,26 @@ def __init__(self, init_value: float, from_min: float, from_max: float, to_min:
             The minimum value for the target interval.
         to_max : float
             The maximum value for the target interval.
+        deadband : float
+            The ratio of input area on both sides of the input center that should be centered in the output.
         """
-        # f(x) = (x - src_min) * (tgt_len / src_len) + tgt_min
-        #      = x * (tgt_len / src_len) + (tgt_min - src_min * (tgt_len / src_len))
-        #      = x * scale + offset
-        self._scale = (to_max - to_min) / (from_max - from_min)
-        self._offset = to_min - from_min * self._scale
+
+        # Set up slopes and output offsets, as well as the bounds for the 3 regions introduced by the deadband
+        self._scale = (to_max - to_min) / ((1 - deadband) * (from_max - from_min))
+        self._offset = (from_max + from_min) / 2
+        input_center = (to_max + to_min) / 2
+        self._deadband_high = deadband * (from_max - input_center) + input_center
+        self._deadband_low = deadband * (from_min - input_center) + input_center
+        
+        # Sort the regions in the event of inversion maps
+        self._low_region_lower_bound = min(self._deadband_low, from_min)
+        self._low_region_upper_bound = max(self._deadband_low, from_min)
+        self._high_region_lower_bound = min(self._deadband_high, from_max)
+        self._high_region_upper_bound = max(self._deadband_high, from_max)
+        self._center_region_lower_bound = min(self._low_region_upper_bound, self._high_region_lower_bound)
+        self._center_region_upper_bound = max(self._low_region_upper_bound, self._high_region_lower_bound)
```

Comment:
Testing on the XBox-360-similar controllers did not work as expected.  Pending further testing.

Reply:
The typo causing this bug is patched in 3c7f592.  Testing with competition controllers is complete; the deadbands and existing controls behave as expected.